frontend/src/main/scala/bloop/engine/ClientPool.scala

```diff
@@ -0,0 +1,38 @@
+package bloop.engine
+
+import com.martiansoftware.nailgun.{NGClientDisconnectReason, NGClientListener, NGContext}
+
+sealed trait CloseEvent
+case object Heartbeat extends CloseEvent
+case object SocketError extends CloseEvent
+case object SocketTimeout extends CloseEvent
+case object SessionShutdown extends CloseEvent
+case object InternalError extends CloseEvent
```

Comment:
Why not using `NGClientDisconnectReason` directly?

Reply:
Because I want not to depend on nailgun data structures as much as possible. Who knows what we'll use in the future. It didn't take a long time to do this, but in my opinion it's worth it :smile: